vertical-pod-autoscaler/pkg/recommender/routines/recommender.go

```diff
@@ -100,15 +100,21 @@ func (r *recommender) UpdateVPAs() {
 		containerResources := make([]vpa_types.RecommendedContainerResources, 0, len(resources))
 		for containerName, res := range resources {
 			containerResources = append(containerResources, vpa_types.RecommendedContainerResources{
-				ContainerName: containerName,
-				Target:        model.ResourcesAsResourceList(res.Target),
-				LowerBound:    model.ResourcesAsResourceList(res.LowerBound),
-				UpperBound:    model.ResourcesAsResourceList(res.UpperBound),
+				ContainerName:  containerName,
+				Target:         model.ResourcesAsResourceList(res.Target),
+				LowerBound:     model.ResourcesAsResourceList(res.LowerBound),
+				UpperBound:     model.ResourcesAsResourceList(res.UpperBound),
+				UncappedTarget: model.ResourcesAsResourceList(res.Target),
```

Comment:
This is confusing even though correct.

Could you please add some explanation what's going on here as a comment?
Or maybe extract full creation of containerResources [up to line 116 or thereabouts] to a function where you could simply comment on its internals w/o overloading the code here.

Reply:
I have extracted to a new method. Reads way better, thanks!